src/main/java/PieChartView.java

```diff
@@ -76,23 +78,34 @@ private void drawPieChartWithLabels(Graphics g) {
             double radians = Math.toRadians(midAngle);
             int labelX = centerXChart + (int) (Math.cos(radians) * labelRadius);
             int labelY = centerYChart - (int) (Math.sin(radians) * labelRadius);
+            AffineTransform originalTransform = g2d.getTransform(); // Translate to label position
+            g2d.translate(labelX, labelY); // Rotate to match slice direction
+            double rotationAngle = -radians; // Default rotation
+            if (midAngle >= 90 && midAngle <= 270) {
+                rotationAngle += Math.PI; // Add 180° (π radians) for labels on the left half
+            }
+            g2d.rotate(rotationAngle); // Adjust rotation direction if needed
 
             // Draw the label centered within the slice
             String label = entry.getKey();
             int textWidth = metrics.stringWidth(label);
             int textHeight = metrics.getHeight();
             g2d.setColor(Color.BLACK); // Text color
-            g2d.drawString(label, labelX - textWidth / 2, labelY + textHeight / 4);
+            //g2d.drawString(label, labelX - textWidth / 2, labelY + textHeight / 4);
```

Comment:
Remove redundant or commented-out code to make the implementation cleaner:
java